src/mono/sample/wasm/wasm.mk

```diff
@@ -10,6 +10,13 @@ CONFIG?=Release
 
 WASM_DEFAULT_BUILD_ARGS?=/p:TargetArchitecture=wasm /p:TargetOS=Browser /p:Configuration=$(CONFIG)
 
+# if we're in a devcontainer, don't try to open the browser
+ifneq ("$(wildcard $(TOP)/artifacts/prebuild.sha)", "")
```

Comment:
Some day I'm hoping to remove this `prebuild.sha` file, if Codespaces adds support for keeping the same commit SHA as what the prebuilt image was built for.

Is there another condition we can use here?

Reply:
There's probably some environment variable that gets set in the container. I'll check the docs...

Reply:
I didn't find any special environment variable from the dev container doc. Another option is to create an environment variable as an indicator for using dev container.

Reply:
In my codespace, using `$ printenv`, the following look promising:

```
 $ printenv
CLOUDENV_ENVIRONMENT_ID=XXX
CODESPACE_NAME=eerhardt-dotnet-runtime-x5gv969qh6xv7
CODESPACES=true
DOTNET_RUNNING_IN_CONTAINER=true
CODESPACE_VSCODE_FOLDER=/workspaces/runtime
```

We could also just check if we are running in a Docker container, which means we can't open a browser. According to https://stackoverflow.com/a/25518538/3680432, we can test for the `/.dockerenv` file to detect.

Reply:
I see that CodeSpace sets a bunch. If the name won't be changed, we could choose one from them.

Reply:
`$CODESPACES` looks like it's meant to be used to detect if we're in codespaces https://github.community/t/codespaces-environment-variable-is-not-set-when-using-docker-compose/139027/2. Switched to that one.

Reply:
Is "are we in codespaces?" the right question to be asking here? Wouldn't the better question be "are we in a Docker container?"

In theory, I could be building/working in a Docker container on my local machine. We wouldn't want to launch the browser in that case, right?